sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -880,6 +860,58 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             }
         }
 
+        private async Task<ServiceBusReceiver> CreateReceiverIfNotExist(
+            string sessionId,
+            ServiceBusReceiverOptions receiverOptions,
+            CancellationToken cancellationToken)
+        {
+            ServiceBusReceiver receiver = null;
+            await receiverCreationLockMap[sessionId].WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
If this fails, the finally won't get hit. This probably is okay, but if someone adds more code under here and before try, then it would cause issues. Can we follow the pattern we use with releasing MaxConcurrentAcceptSessionsSemaphore